src/config.js

```diff
@@ -954,24 +963,58 @@ class Config {
       const scopedSettings = newSettings
       newSettings = newSettings['*']
       delete scopedSettings['*']
-      this.resetUserScopedSettings(scopedSettings)
+      this.resetScopedSettings(scopedSettings, {source})
     }
 
     return this.transact(() => {
-      this.settings = {}
+      this._clearUnscopedSettingsForSource(source)
       this.settingsLoaded = true
-      for (let key in newSettings) { const value = newSettings[key]; this.set(key, value, {save: false}) }
+      for (let key in newSettings) {
+        const value = newSettings[key]
+        this.set(key, value, {save: false, source})
+      }
       if (this.pendingOperations.length) {
         for (let op of this.pendingOperations) { op() }
         this.pendingOperations = []
       }
     })
   }
 
+  _clearUnscopedSettingsForSource (source) {
+    switch (source) {
```

Comment:
nit: does this really need to be a switch? feels like an if/else would do.

Reply:
I left it as a switch just in case more source settings are added in the future (which might need to happen for path specific configs)- but since there are only two an if-else would definitely do!